.github/workflows/clear-runs.yml

```diff
@@ -0,0 +1,83 @@
+name: Clear outdated runs
+
+on: workflow_call
+
+jobs:
+  get-previous-runs:
+    name: Get previous runs
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Get current workflow id
+        id: get-workflow-id
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const response = await github.rest.actions.listRepoWorkflows({
+               owner: context.repo.owner,
+               repo: context.repo.repo
+            })
+            console.log("Get all workflows response = ", response.data);
+            const workflows = response.data.workflows.filter((workflow) => (workflow.name === "${{ github.workflow }}"));
+            console.log("Current workflow ID = ", workflows[0].id);
+            return workflows[0].id;
+
+      - name: Get pervious waiting workflow runs
+        id: get-waiting-runs
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const response = await github.rest.actions.listWorkflowRuns({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               workflow_id: ${{ steps.get-workflow-id.outputs.result }},
+               branch: "main",
```

Comment:
This should be the current branch instead of hardcoded, avoids complication if the `main` branch is ever renamed or for edge cases like deploying from alpha branches.